test/Calibur.isValidSignature.t.sol

```diff
@@ -349,20 +372,108 @@ contract CaliburIsValidSignatureTest is DelegationHandler, HookHandler, ERC1271H
         signerAccount.isValidSignature(digest, wrappedSignature);
     }
 
+    function test_isValidSignature_WebAuthnP256_wrappedSignature___typedDataSign_expiredKey_reverts() public {
+        TestKey memory webAuthnP256Key = TestKeyManager.initDefault(KeyType.WebAuthnP256);
+        bytes memory signature = webAuthnP256Key.sign(TEST_TYPED_DATA_SIGN_DIGEST);
+        (bytes32 appDomainSeparator, string memory contentsDescr, bytes32 contentsHash) = getERC1271Fixtures();
+        bytes memory typedDataSignSignature =
+            TypedDataSignBuilder.buildTypedDataSignSignature(signature, appDomainSeparator, contentsHash, contentsDescr);
+        bytes memory wrappedSignature = abi.encode(webAuthnP256Key.toKeyHash(), typedDataSignSignature, EMPTY_HOOK_DATA);
+
+        vm.warp(100);
+        Settings keySettings = SettingsBuilder.init().fromExpiration(uint40(block.timestamp - 1));
+
+        vm.startPrank(address(signerAccount));
+        signerAccount.register(webAuthnP256Key.toKey());
+        signerAccount.update(webAuthnP256Key.toKeyHash(), keySettings);
+        vm.stopPrank();
+
+        bytes32 digest = mockERC1271VerifyingContract.hashTypedDataV4(contentsHash);
+        vm.expectRevert(abi.encodeWithSelector(IKeyManagement.KeyExpired.selector, uint40(block.timestamp - 1)));
+        vm.prank(address(mockERC1271VerifyingContract));
+        signerAccount.isValidSignature(digest, wrappedSignature);
+    }
+
     /**
      *
      * MARK: Invalid signer tests
      *
      */
 
     /**
-     * Scenario: Root key
-     * 1. Not typed data sign
-     * 2. Not nested personal sign
-     * 3. Not safe ERC1271 caller
+     * Scenario: SECP256K1 key
+     * 1. Raw signature not from the root key
      * = invalid signer
      */
-    function test_isValidSignature_rootKey_notTypedDataSign_invalidSigner() public {
+    function test_isValidSignature_secp256k1_rawSignature_reverts_withKeyDoesNotExist() public {
+        TestKey memory key = TestKeyManager.withSeed(KeyType.Secp256k1, 0xb0b);
+        vm.prank(address(signer));
+        signerAccount.register(key.toKey());
+
+        bytes32 digest = keccak256("test");
+        bytes memory signature = key.sign(digest);
+
+        vm.expectRevert(IKeyManagement.KeyDoesNotExist.selector);
+        signerAccount.isValidSignature(digest, signature);
+    }
+
+    /// P256 signatures are 64 bytes, so the first 32 bytes will be assumed to be the keyHash which will revert with KeyDoesNotExist()
```

Comment:
I like the test tags! Could we add them here too? I think this one would be

    /**
     * Scenario: P256 key
     * 1. Raw signature
     * 2. Not typed data sign
     * 3. Not nested personal sign
     * = revert 
     */


?

Reply:
With your suggestion above to return false early if it is 64 or 65 but not from the root key this would be:
```
    /**
     * Scenario: P256 key
     * 1. 64 or 65-byte signature not from the root key
     * = invalid signer
     */
```

Reply:
yep that makes sense